File: web/src/plugin/tools/api/sendChat.js
import service from '@/utils/request'

export const createSendChatTask = (data) => {
  return service({
    url: '/toolsSendChat/createSendChatTask',
    method: 'post',
    data
  })
}

export const getSendChatTaskList = (params) => {
  return service({
    url: '/toolsSendChat/getSendChatTaskList',
    method: 'get',
    params
  })
}

export const stopSendChatTask = (params) => {
  return service({
    url: '/toolsSendChat/stopSendChatTask',
    method: 'put',
    params
  })
}

File: server/plugin/tools/router/send_chat.go
package router

import (
	"github.com/gin-gonic/gin"
)

type sendChat struct{}

func (r *sendChat) Init(public, private *gin.RouterGroup) {
	group := private.Group("toolsSendChat")
	group.POST("createSendChatTask", apiInfo.SendChat.CreateSendChatTask)
	group.GET("getSendChatTaskList", apiInfo.SendChat.GetSendChatTaskList)
	group.PUT("stopSendChatTask", apiInfo.SendChat.StopSendChatTask)
}
